packages/convex-helpers/server/pagination.ts

```diff
@@ -0,0 +1,301 @@
+import { Value, convexToJson } from "convex/values";
+import {
+  DocumentByName,
+  GenericDataModel,
+  GenericDatabaseReader,
+  IndexNames,
+  NamedTableInfo,
+  SchemaDefinition,
+  TableNamesInDataModel,
+} from "convex/server";
+
+export type IndexKey = Value[];
+
+export type PageRequest<
+  DataModel extends GenericDataModel,
+  T extends TableNamesInDataModel<DataModel>,
+> = {
+  /// Request a page of documents from this table.
+  table: T;
+  /// Where the page starts. Default or empty array is the start of the table.
+  startIndexKey?: IndexKey;
+  /// Whether the startIndexKey is inclusive. Default is false.
```

Comment:
Is there a standard here? I would have expected the flip (inclusive start, exclusive end).

Reply:
when you do it like this, you can combine and refetch pages easily, using the index keys of the returned documents. 

Like if i fetch the first page and get results 1 to 10, i can refetch it by quering the range (0, 10] and i can fetch the next page as (10, infinity]. If you make the end exclusive, you need to get an index key >10 to refetch the first page, which you don't have if you only have documents 1 to 10.

Reply:
ah yes, that makes a lot of sense, thanks